backend/tests/share.test.ts

```diff
@@ -0,0 +1,93 @@
+import { assertDBValues, seedData, truncateTable } from './utils'
+import request from 'supertest'
+import { sequelize, sequelizeLoader } from '../src/db/loader'
+import app from '../src'
+import { AuthMiddleware } from '../src/middlewares/auth.middleware'
+const req = request(app)
+
+describe('Code Share Test', () => {
+	beforeAll(async () => {
+		await sequelizeLoader()
+		AuthMiddleware.decodeJWTHeader = jest.fn(async (val) => val)
+	})
+
+	beforeEach(async () => {
+		await truncateTable()
+	})
+
+	afterAll(async () => {
+		await sequelize.close()
+	})
+
+	const defaultReq = {
+		code: "print('hello world')",
+		input: '',
+	}
+
+	const defaultRes = {
+		uid: '1',
+		codename: 'sample',
+		...defaultReq,
+		shareId: null,
+	}
+
+	test('Sanity Check', async () => {
+		await seedData([defaultRes])
+		await req
+			.post('/user/codes/sample/share')
+			.set('Authorization', '1')
+			.expect(201)
+			.then(async (res) => {
+				const shareId = res.body.shareId
+				await assertDBValues([
+					{
+						...defaultRes,
+						shareId,
+					},
+				])
+				req
+					.get('/codes?id=' + shareId)
+					.expect(200)
+					.expect(defaultReq)
+			})
+		await req
+			.post('/user/codes/sample/share')
+			.set('Authorization', '1')
+			.expect(201)
+			.then(async (res) => {
+				const shareId = res.body.shareId
```

Comment:
Can add an extra check here that the shareId has changed? In case recreating the link didn't work.

Reply:
Updated